pkg/utils/dataset/lifecycle/node.go

```diff
@@ -167,6 +171,17 @@ func LabelCacheNode(nodeToLabel v1.Node, runtimeInfo base.RuntimeInfoInterface,
 			updatedLabels = append(updatedLabels, exclusiveLabel)
 		}
 
+		if currentDatasetNum, ok := toUpdate.Labels[datasetLabel]; ok {
+			currentData, err := strconv.Atoi(currentDatasetNum)
+			if err != nil {
+				return err
+			}
+			toUpdate.Labels[datasetLabel] = strconv.Itoa(currentData + 1)
+		} else {
+			toUpdate.Labels[datasetLabel] = "1"
+		}
+		updatedLabels = append(updatedLabels, datasetLabel)
+
 		labelNodeWithCapacityInfo(toUpdate, runtimeInfo, &updatedLabels)
 
 		err = client.Update(context.TODO(), toUpdate)
```

Comment:
I think we should we `RetryOnConflict` here if the update failed because of resource version conflicts